tests/test_shadow_run_id.py

```
@@ -0,0 +1,109 @@
+        # run selector, or a rehearsal's zero fills read as a live session's.
+        rid = shadow_run_id()
+        assert rid.startswith("shadow-")
+        assert not rid.startswith("run-")
+
+    def test_an_explicit_override_is_honoured(self):
+        # The operator continuing one rehearsal across a restart is the one
+        # case where sharing an id is the point.
+        with mock.patch.dict(os.environ, {"SH_RUN_ID": "shadow-deadbeef"}):
+            assert shadow_run_id() == "shadow-deadbeef"
+
+    def test_the_live_lock_file_is_never_consulted(self, tmp_path, monkeypatch):
```

Ruff (0.16.2)

[warning] Unused method argument: `tmp_path`

(ARG002)